src/fides/api/models/sql_models.py

```diff
@@ -514,6 +532,124 @@ def create(
+        """Returns the instance-level indication of whether the purpose should be included"""
+        query: Select = select([TCFPublisherOverride.is_included]).where(
+            TCFPublisherOverride.purpose == self.purpose
+        )
+        async_session: AsyncSession = async_object_session(self)
+        async with async_session.begin():
+            result = await async_session.execute(query)
+            return result.scalars().first()
+
+    @_publisher_override_is_included_join.expression
+    def _publisher_override_is_included_join(cls) -> ScalarSelect:
+        """Returns the class-level indication of whether the purpose should be included"""
+        return (
+            select([TCFPublisherOverride.is_included])
+            .where(TCFPublisherOverride.purpose == cls.purpose)
+            .as_scalar()
+        )
+
+    @hybrid_property
+    async def overridden_legal_basis_for_processing(self) -> Optional[str]:
+        """
+        Instance-level override of the legal basis for processing based on
+        publisher preferences.
+        """
+        if not (
+            CONFIG.consent.override_vendor_purposes
+            and self.flexible_legal_basis_for_processing
+        ):
+            return self.legal_basis_for_processing
+
+        is_included: Optional[bool] = await self._publisher_override_is_included_join
+
+        if is_included is False:
+            # Overriding to False to match behavior of class-level override.
+            # Class-level override of legal basis to None removes Privacy Declaration
+            # from Experience
+            return None
+
+        overridden_legal_basis: Optional[str] = (
+            await self._publisher_override_legal_basis_join
+        )
+
+        return (
+            overridden_legal_basis
+            if overridden_legal_basis  # pylint: disable=using-constant-test
+            else self.legal_basis_for_processing
+        )
+
+    @overridden_legal_basis_for_processing.expression
+    def overridden_legal_basis_for_processing(
+        cls,
+    ) -> Union[InstrumentedAttribute, Case]:
+        """
+        Class-level override of the legal basis for processing based on
+        publisher preferences.
+        """
+        if not CONFIG.consent.override_vendor_purposes:
+            return cls.legal_basis_for_processing
+
+        return case(
+            [
+                (
+                    cls.flexible_legal_basis_for_processing.is_(False),
+                    cls.legal_basis_for_processing,
+                ),
+                (
+                    cls._publisher_override_is_included_join.is_(False),
+                    None,
+                ),
+                (
+                    cls._publisher_override_legal_basis_join.is_(None),
+                    cls.legal_basis_for_processing,
+                ),
+            ],
+            else_=cls._publisher_override_legal_basis_join,
+        )
```

Comment:
This is the primary query change that is used to update the Privacy Declaration query which downstream affects the contents of the TCF Experience.   The `@overridden_legal_basis_for_processing.expression` denotes it as a class-level hybrid property.  

This `overridden_legal_basis_for_processing` becomes the legal basis that is filtered for all intents and purposes downstream.  So if we shouldn't override it, we default to the regular legal basis. If it shouldn't be included at all, we default to None.  If we should override it, we return the legal basis specified in the override table.

Reply:
ok, i think i've got it. let's definitely include this behavior in the docstring for the method and/or other places.

it is a bit counterintuitive to me that the general "exclusion" signal is coming through on the `overridden_legal_basis_for_processing` property, since the "exclusion" applies to the purpose (`PrivacyDeclaration`) as a whole, not its legal basis. but i understand that this is a pretty _efficient_ way to represent the different states here, and it plugs in seamlessly to existing downstream code. i'm just a bit concerned about maintainability, i.e. that we may be hiding some behavior by overloading this property. this is related to my comment below.

i'm still thinking through this! maybe i'm not conceptualizing it properly.

Reply:
In my latest draft, I've removed all the hybrid properties except for "purpose" which I think still makes sense. I'm instead joining the primary privacy declaration query with a new subquery that is very similar to this logic.  So less is living on the model now but the query generated by SQLAlchemy is the one I'd expect.

Reply:
I still think capturing everything in one place, combining the different states into one new attribute makes sense for the TCF experience, going along with our original rule that a declaration needs a consent or LI legal basis to show up at all.  Because this logic no longer lives on the model, I'm not as concerned about overloading it, since this is now written very specifically for the TCF overlay.